airbyte_cdk/cli/airbyte_cdk/exceptions.py

```diff
@@ -0,0 +1,33 @@
+# Copyright (c) 2025 Airbyte, Inc., all rights reserved.
+"""Exceptions for the Airbyte CDK CLI."""
+
+from dataclasses import dataclass
+from typing import List
+
+from airbyte_cdk.sql.exceptions import AirbyteConnectorError
+
+
+@dataclass
+class ConnectorSecretWithNoValidVersionsError(AirbyteConnectorError):
+    """Error when a connector secret has no valid versions."""
+
+    connector_name: str
+    secret_names: List[str] = None  # type: ignore
+    connector_secret_urls: List[str] = None  # type: ignore
+    
+    def __post_init__(self):
+        """Initialize default values for lists."""
+        super().__post_init__()
+        if self.secret_names is None:
+            self.secret_names = []
+        if self.connector_secret_urls is None:
+            self.connector_secret_urls = []
```

Comment:
Scrap this post-init. It's unnecessary complexity. Remove the "= None" default (aka make required) and also use `@dataclass(kwargs=True)` to force callers to explicitly pass their args.